File: easy-common/src/main/java/xyz/srclab/common/base/Defaults.java
package xyz.srclab.common.base;

import kotlin.text.Charsets;

import java.nio.charset.Charset;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class Defaults {

    public static final String LINE_SEPARATOR = System.lineSeparator();

    public static final Locale LOCALE = Locale.US;

    public static final Charset CHARSET = Charsets.UTF_8;

    public static final TimeUnit TIME_UNIT = TimeUnit.SECONDS;
}

File: boat-core/src/main/kotlin/xyz/srclab/common/base/Defaults.kt
@file:JvmName("Defaults")

package xyz.srclab.common.base

import java.io.File
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets





/**
 * Default concurrent level: Math.min([Systems.availableProcessors] * 2, 16).
 */
fun default: Int
    get() {
        return (Systems.availableProcessors * 2).coerceAtMost(16)
    }

/**
 * Default file separator: [File.separator].
 * "/" on Unix, "\\" on Windows.
 */
@JvmField
val FILE_SEPARATOR: String = File.separator

/**
 * Default path separator: [File.pathSeparator].
 * ":" on Unix, ";" on Windows.
 */
@JvmField
val PATH_SEPARATOR: String = File.pathSeparator

/**
 * Default line separator: [System.lineSeparator].
 * "\n" on Unix, "\r\n" on Windows, "\r" on Mac.
 */
@JvmField
val LINE_SEPARATOR: String = System.lineSeparator()